include/aotriton/_internal/triton_kernel.h

```diff
@@ -24,11 +27,24 @@ class TritonKernel {
   void clear_decompressed_image();
 #endif
 private:
+  std::tuple<hipFunction_t, hipError_t> load_for_device(int device_id, const char* kernel_name);
+  hipFunction_t cfind_function(int device_id) const;
+
   const void* kernel_image_ = nullptr;
   size_t image_size_ = 0;
+  struct DeviceFunction {
+    DeviceFunction(int device_id_,
+                   hipModule_t mod_,
+                   hipFunction_t func_);
+    ~DeviceFunction();
+    int device_id = -1;
```

Comment:
This device_id is not used later on. And you already have a unordered_map whose key is device_id

Reply:
It's kept here in case of possible sanity checks.